api/base/filters.py

```diff
@@ -44,35 +79,17 @@ class ODMFilterMixin(object):
 
     # TODO Handle simple and complex non-standard fields
 
-    TRUTHY = set(['true', 'True', 1, '1'])
-    FALSY = set(['false', 'False', 0, '0'])
-    DEFAULT_OPERATOR = 'eq'
-
-    # For the field_comparison_operators, instances can be a class or a tuple of classes
-    field_comparison_operators = [
-        {
-            'field_type': ser.CharField,
-            'comparison_operator': 'icontains'
-        },
-        {
-            'field_type': ser.ListField,
-            'comparison_operator': 'in'
-        }
-    ]
+    field_comparison_operators = {
+        ser.CharField: 'icontains',
+        ser.ListField: 'in',
+    }
 
     def get_comparison_operator(self, key):
-
-        for operator in self.field_comparison_operators:
-            if isinstance(self.serializer_class._declared_fields[key], operator['field_type']):
-                return operator['comparison_operator']
-
-        return self.DEFAULT_OPERATOR
-
-    def is_filterable_field(self, key):
-        try:
-            return key.strip() in self.serializer_class.filterable_fields
-        except AttributeError:
-            return key.strip() in self.serializer_class._declared_fields
+        field_type = type(self.serializer_class._declared_fields[key])
```

Comment:
Same as above. If the mixin only works when certain assertions about the inheriting class hold true, I'd like to see those assertions made in the mixin's constructor.